### PR TITLE
Fixed integer range minus bugs

### PR DESCRIPTION
**Pull request by staabm**
Fixes Integer-Range-Minus bugs mentioned in https://github.com/phpstan/phpstan/issues/5614

### CODE REVIEW

File: tests/PHPStan/Analyser/data/integer-range-types.php [hunk 1]
```diff
@@ -254,16 +255,18 @@ public function math($i, $j, $z, $pi, $r1, $r2, $rMin, $rMax, $x, $y) {
 		assertType('float|int<1, max>', $rMax / 4);
 
 		assertType('int<6, 20>', $r1 + $r2);
-		assertType('int<-4, 0>', $r1 - $r2);
+		assertType('int<-9, 5>', $r1 - $r2);
 		assertType('int<5, 100>', $r1 * $r2);
 		assertType('float|int<0, 1>', $r1 / $r2);
 
+		assertType('int<-99, 19>', $r1 - $r3);
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/integer-range-types.php:262]: this line is covering https://phpstan.org/r/0f2bf43b-e5a6-436b-ac44-e62a2c60ae94 from  "Wrong inference with Minus" from https://github.com/phpstan/phpstan/issues/5614

File: tests/PHPStan/Analyser/data/integer-range-types.php [hunk 2]
```diff
@@ -254,16 +255,18 @@ public function math($i, $j, $z, $pi, $r1, $r2, $rMin, $rMax, $x, $y) {
 		assertType('float|int<1, max>', $rMax / 4);
 
 		assertType('int<6, 20>', $r1 + $r2);
-		assertType('int<-4, 0>', $r1 - $r2);
+		assertType('int<-9, 5>', $r1 - $r2);
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/integer-range-types.php:258]: fixed wrong expectation, see https://www.wolframalpha.com/input/?i=%28interval%5B1%2C10%5D++-+interval%5B5%2C10%5D%29

File: tests/PHPStan/Analyser/data/integer-range-types.php [hunk 3]
```diff
@@ -254,16 +255,18 @@ public function math($i, $j, $z, $pi, $r1, $r2, $rMin, $rMax, $x, $y) {
...
 
+		assertType('int<-99, 19>', $r1 - $r3);
+
 		assertType('int<min, 15>', $r1 + $rMin);
 		assertType('int<min, 5>', $r1 - $rMin);
 		assertType('int<min, 50>', $r1 * $rMin);
 		assertType('float|int<min, 2>', $r1 / $rMin);
 		assertType('int<min, 15>', $rMin + $r1);
-		assertType('int<min, -5>', $rMin - $r1);
+		assertType('int<-5, max>', $rMin - $r1);
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/integer-range-types.php:269]: fixed wrong expectation, see https://www.wolframalpha.com/input/?i=%28interval%5B%E2%88%9E%2C5%5D++-+interval%5B1%2C10%5D%29

File: tests/PHPStan/Analyser/data/integer-range-types.php [hunk 4]
```diff
@@ -272,7 +275,7 @@ public function math($i, $j, $z, $pi, $r1, $r2, $rMin, $rMax, $x, $y) {
 		assertType('int<5, max>', $r1 * $rMax);
 		assertType('float|int<0, max>', $r1 / $rMax);
 		assertType('int<6, max>', $rMax + $r1);
-		assertType('int<4, max>', $rMax - $r1);
+		assertType('int<-5, max>', $rMax - $r1);
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/integer-range-types.php:278]: fixed wrong expectation, see https://www.wolframalpha.com/input/?i=%28interval%5B5%2C%E2%88%9E%5D++-+interval%5B1%2C10%5D%29

File: tests/PHPStan/Analyser/data/math.php [hunk 5]
```diff
@@ -52,7 +52,7 @@ public function doBaz(int $rangeFiveBoth, int $rangeFiveLeft, int $rangeFiveRigh
 		assertType('int', $rangeFiveRight - $rangeFiveLeft);
 
 		assertType('int<-10, 10>', $rangeFiveBoth + $rangeFiveBoth);
-		assertType('0', $rangeFiveBoth - $rangeFiveBoth);
+		assertType('int<-10, 10>', $rangeFiveBoth - $rangeFiveBoth);
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/math.php:55]: https://www.wolframalpha.com/input/?i=%28interval%5B-5%2C5%5D+-+interval%5B-5%2C5%5D++%29

File: tests/PHPStan/Analyser/data/math.php [hunk 6]
```diff
@@ -61,10 +61,10 @@ public function doBaz(int $rangeFiveBoth, int $rangeFiveLeft, int $rangeFiveRigh
 		assertType('int<min, 0>', $rangeFiveBoth - $rangeFiveRight);
 
 		assertType('int<-10, max>', $rangeFiveLeft + $rangeFiveBoth);
-		assertType('int<0, max>', $rangeFiveLeft - $rangeFiveBoth);
+		assertType('int<-10, max>', $rangeFiveLeft - $rangeFiveBoth);
 
 		assertType('int<min, 10>', $rangeFiveRight + $rangeFiveBoth);
-		assertType('int<min, 0>', $rangeFiveRight - $rangeFiveBoth);
+		assertType('int<0, max>', $rangeFiveRight - $rangeFiveBoth);
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/math.php:67]: https://www.wolframalpha.com/input/?i=%28interval%5B%E2%88%9E%2C5%5D+-+interval%5B-5%2C5%5D++%29

File: tests/PHPStan/Analyser/data/math.php [hunk 7]
```diff
@@ -61,10 +61,10 @@ public function doBaz(int $rangeFiveBoth, int $rangeFiveLeft, int $rangeFiveRigh
 		assertType('int<min, 0>', $rangeFiveBoth - $rangeFiveRight);
 
 		assertType('int<-10, max>', $rangeFiveLeft + $rangeFiveBoth);
-		assertType('int<0, max>', $rangeFiveLeft - $rangeFiveBoth);
+		assertType('int<-10, max>', $rangeFiveLeft - $rangeFiveBoth);
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/math.php:64]: https://www.wolframalpha.com/input/?i=%28interval%5B-5%2C%E2%88%9E%5D++-+interval%5B-5%2C5%5D%29

**Comment by staabm** [comment]
should be good to go. remaining test-errors are unrelated.

//cc @orklah

**Comment by staabm** [comment]
> Note the -∞ instead of ∞

oh, I need to double check the new expectations..

File: tests/PHPStan/Analyser/data/math.php [hunk 8]
```diff
@@ -61,10 +61,10 @@ public function doBaz(int $rangeFiveBoth, int $rangeFiveLeft, int $rangeFiveRigh
...
 
 		assertType('int<min, 10>', $rangeFiveRight + $rangeFiveBoth);
-		assertType('int<min, 0>', $rangeFiveRight - $rangeFiveBoth);
+		assertType('int<min, 10>', $rangeFiveRight - $rangeFiveBoth);
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/math.php:67]: https://www.wolframalpha.com/input/?i=%28interval%5B-%E2%88%9E%2C5%2C%5D++-+interval%5B-5%2C5%5D%29

File: tests/PHPStan/Analyser/data/integer-range-types.php [hunk 9]
```diff
@@ -254,16 +255,18 @@ public function math($i, $j, $z, $pi, $r1, $r2, $rMin, $rMax, $x, $y) {
...
 
+		assertType('int<-99, 19>', $r1 - $r3);
+
 		assertType('int<min, 15>', $r1 + $rMin);
 		assertType('int<min, 5>', $r1 - $rMin);
 		assertType('int<min, 50>', $r1 * $rMin);
 		assertType('float|int<min, 2>', $r1 / $rMin);
 		assertType('int<min, 15>', $rMin + $r1);
-		assertType('int<min, -5>', $rMin - $r1);
+		assertType('int<min, 4>', $rMin - $r1);
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/integer-range-types.php:269]: https://www.wolframalpha.com/input/?i=interval%5B-%E2%88%9E%2C5%5D++-+interval%5B1%2C10%5D

File: tests/PHPStan/Analyser/data/math.php [hunk 10]
```diff
@@ -52,19 +52,19 @@ public function doBaz(int $rangeFiveBoth, int $rangeFiveLeft, int $rangeFiveRigh
...
 		assertType('int<-10, 10>', $rangeFiveBoth + $rangeFiveBoth);
-		assertType('0', $rangeFiveBoth - $rangeFiveBoth);
+		assertType('int<-10, 10>', $rangeFiveBoth - $rangeFiveBoth);
 
 		assertType('int<-10, max>', $rangeFiveBoth + $rangeFiveLeft);
 		assertType('int', $rangeFiveBoth - $rangeFiveLeft);
 
 		assertType('int<min, 10>', $rangeFiveBoth + $rangeFiveRight);
-		assertType('int<min, 0>', $rangeFiveBoth - $rangeFiveRight);
+		assertType('int<-10, max>', $rangeFiveBoth - $rangeFiveRight);
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/math.php:61]: https://www.wolframalpha.com/input/?i=interval%5B-5%2C5%5D+-+interval%5B-%E2%88%9E%2C+5%5D+++

File: tests/PHPStan/Analyser/data/integer-range-types.php [hunk 11]
```diff
@@ -254,16 +255,18 @@ public function math($i, $j, $z, $pi, $r1, $r2, $rMin, $rMax, $x, $y) {
...
-		assertType('int<-4, 0>', $r1 - $r2);
+		assertType('int<-9, 5>', $r1 - $r2);
 		assertType('int<5, 100>', $r1 * $r2);
 		assertType('float|int<0, 1>', $r1 / $r2);
 
+		assertType('int<-99, 19>', $r1 - $r3);
+
 		assertType('int<min, 15>', $r1 + $rMin);
-		assertType('int<min, 5>', $r1 - $rMin);
+		assertType('int<-4, max>', $r1 - $rMin);
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/integer-range-types.php:265]: https://www.wolframalpha.com/input/?i=interval%5B1%2C10%5D+-+interval%5B-%E2%88%9E%2C+5%5D+++

**Comment by staabm** [comment]
should be good to go. remaining test-errors are unrelated.

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**